bin/mympirun.py

```diff
@@ -40,83 +40,139 @@
-                               "of the direct calls or make sure the mpirun command can be found. "
-                               "Found MPI %s") % (scriptname, ismpirun, ", ".join(found_mpi_names)))
+        mo.log.raiseException(
+            ("No MPI class found that supports scriptname %s; ismpirun %s). "
+             "Please use mympirun through one of the direct calls or make sure "
+             "the mpirun command can be found. Found MPI %s") %
+            (scriptname, ismpirun, ", ".join(found_mpi_names)))
     else:
-        mo.log.debug("Found MPI class %s (scriptname %s; ismpirun %s)" % (mpi.__name__, scriptname, ismpirun))
+        mo.log.debug("Found MPI class %s (scriptname %s; ismpirun %s)" %
+                     (mpi.__name__, scriptname, ismpirun))
 
     if sched is None:
-        mo.log.raiseException("No sched class found (options.schedtype %s ; found Sched classes %s)" %
-                              (mo.options.schedtype, ", ".join(found_sched_names)))
+        mo.log.raiseException(
+            ("No sched class found (options.setsched %s ; found Sched "
+             "classes %s)") % (mo.options.setsched,
+             ", ".join(found_sched_names)))
     else:
-        mo.log.debug("Found sched class %s from options.schedtype %s (all Sched found %s)" %
-                     (sched.__name__, mo.options.schedtype, ", ".join(found_sched_names)))
+        mo.log.debug(
+            ("Found sched class %s from options.setsched %s (all Sched "
+             "found %s)") % (sched.__name__, mo.options.setsched,
+             ", ".join(found_sched_names)))
 
     return mpi, sched, mo
 
 
+def import_implementations(module):
+    """searches and imports python files in the some folder as a module
+
+    Arguments:
+        module      --  the module
+    """
+
+    _logger.debug("mympirun.py - import_implementations()")
+
+    # get the paths of all the python files in the module folder
+    modulepaths = glob.glob(os.path.dirname(module.__file__) + "/*.py")
+
+    # parse the module namespace
+    namespace = ".".join(module.__name__.split('.')[:-1])
+
+    # transform the paths to module names while discarding __init__.py
+    modulenames = [namespace + "." + os.path.basename(os.path.splitext(f)[0])
+                   for f in modulepaths if os.path.isfile(f) and
+                   "__init__" not in f]
+
+    _logger.debug("namespace: %s", namespace)
+
+    # import the modules
+    modules = map(__import__, modulenames)
+    _logger.debug("mympirun.py - imported modules: %s", modulenames)
+
+    return
+
+
 def main():
     """Main function"""
+    _logger.debug("mympirun.py - main()")
     try:
         m = getinstance(*get_mpi_and_sched_and_options())
         m.main()
-        ec = 0
-    except ExitException:
-        ec = 0
-    except:
-        # # TODO: cleanup, only catch known exceptions
+        sys.exit(0)
+    except Exception, e:
         if os.environ.get('MYMPIRUN_MAIN_EXCEPTION', 0) == '1':
```

Comment:
does this have a function? It should go. 

If there's an exception, we print and exit.